File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/preprocessor/UsingDefinitionInfo.java
package org.cqframework.cql.cql2elm.preprocessor;

import org.cqframework.cql.gen.cqlParser;

public class UsingDefinitionInfo extends BaseInfo {
    private String namespaceName;
    private String name;
    private String version;
    private String localName;

    public String getNamespaceName() {
        return namespaceName;
    }

    public void setNamespaceName(String value) {
        namespaceName = value;
    }

    public String getName() {
        return name;
    }

    public void setName(String value) {
        name = value;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String value) {
        version = value;
    }

    public String getLocalName() {
        return localName;
    }

    public void setLocalName(String value) {
        localName = value;
    }

    public UsingDefinitionInfo withNamespaceName(String value) {
        setNamespaceName(value);
        return this;
    }

    public UsingDefinitionInfo withName(String value) {
        setName(value);
        return this;
    }

    public UsingDefinitionInfo withVersion(String value) {
        setVersion(value);
        return this;
    }

    public UsingDefinitionInfo withLocalName(String value) {
        setLocalName(value);
        return this;
    }

    @Override
    public cqlParser.UsingDefinitionContext getDefinition() {
        return (cqlParser.UsingDefinitionContext)super.getDefinition();
    }

    public void setDefinition(cqlParser.UsingDefinitionContext value) {
        super.setDefinition(value);
    }

    public UsingDefinitionInfo withDefinition(cqlParser.UsingDefinitionContext value) {
        setDefinition(value);
        return this;
    }
}

File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/preprocessor/UsingDefinitionInfo.kt
package org.cqframework.cql.cql2elm.preprocessor

import org.cqframework.cql.gen.cqlParser.UsingDefinitionContext

class UsingDefinitionInfo : BaseInfo() {
    var namespaceName: String? = null
    @JvmField var name: String? = null
    @JvmField var version: String? = null
    var localName: String? = null

    fun withNamespaceName(value: String?): UsingDefinitionInfo {
        namespaceName = value
        return this
    }

    fun withName(value: String?): UsingDefinitionInfo {
        name = value
        return this
    }

    fun withVersion(value: String?): UsingDefinitionInfo {
        version = value
        return this
    }

    fun withLocalName(value: String?): UsingDefinitionInfo {
        localName = value
        return this
    }

    override var definition: UsingDefinitionContext? = null

    fun withDefinition(value: UsingDefinitionContext?): UsingDefinitionInfo {
        this.definition = value
        return this
    }
}
